cmd/scheduler-builder/cmd/root.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"k8s.io/release/pkg/log"
+	"sigs.k8s.io/yaml"
+)
+
+// PatchSchedule main struct to hold the schedules
+type PatchSchedule struct {
+	Schedules []Schedules `yaml:"schedules"`
+}
+
+// PreviousPatches struct to define the old pacth schedules
+type PreviousPatches struct {
+	Release            string `yaml:"release"`
+	CherryPickDeadline string `yaml:"cherryPickDeadline"`
+	TargetDate         string `yaml:"targetDate"`
+}
+
+// Schedules struct to define the release schedule for a specific version
+type Schedules struct {
+	Release            string            `yaml:"release"`
+	Next               string            `yaml:"next"`
+	CherryPickDeadline string            `yaml:"cherryPickDeadline"`
+	TargetDate         string            `yaml:"targetDate"`
+	EndOfLifeDate      string            `yaml:"endOfLifeDate"`
+	PreviousPatches    []PreviousPatches `yaml:"previousPatches"`
+}
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:               "scheduler-builder --schedule-path path/to/schedule.yaml",
+	Short:             "scheduler-builder generate a humam readable format of the Kubernetes release schedule",
+	Example:           "scheduler-builder --schedule-path /home/user/kubernetes/sig-release/releases/schedule.yaml",
+	SilenceUsage:      true,
+	SilenceErrors:     true,
+	PersistentPreRunE: initLogging,
+	RunE: func(*cobra.Command, []string) error {
+		return run(opts)
+	},
+}
+
+type options struct {
+	schedulePath string
+	logLevel     string
+}
+
+var opts = &options{}
+
+var (
+	schedulePathFlag = "schedule-path"
```

Comment:
Could we do `--config`, `--config-path`, or `--config-file` instead?
Do we also want to add `--output-file`, so we can choose where the markdown file is written to?

Reply:
added the output-file flag